src/reader.rs

```diff
@@ -204,21 +146,152 @@ impl Reader {
 
         DbIterator::new(range.clone(), mem_iter, l0_iters, sr_iters).await
     }
+}
 
-    fn unwrap_value_if_not_expired(entry: &RowEntry, now_ttl: i64) -> Option<Bytes> {
-        if is_not_expired(entry, now_ttl) {
-            entry.value.as_bytes()
-        } else {
-            None
+struct LevelGet<'a> {
```

Comment:
why wrap this up in a struct vs standalone methods?

Reply:
a struct may help expose mockable fields, and it follows a single responsibility. it can focus on the dependencies that `get()` is interested in, may help making tests around it easier.